### PR TITLE
Add command to make a connection pooling attachment

### PR DESCRIPTION
**Pull request by coreypurcell**
Add command to create an attachment to a DB that uses the URL for pgbouncer. This is to avoid needing something like a build pack to create the config vars with the pgbouncer URL. 

Work to be completed:

* [x] more tests to cover the case where a credential is specified
* [x] query the api to ensure the DB actually has PGbouncer installed

### CODE REVIEW

**Comment by msakrejda** [comment]
Thanks! At a glance, this looks fine, but I'ld like to give it a more thorough review. I'll try to do that by Monday.

File: commands/connection_pooling.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    {success: false},
+    co(function * () {
+      let releases = yield heroku.get(`/apps/${app}/releases`, {
+        partial: true,
+        headers: { 'Range': 'version ..; max=1, order=desc' }
+      })
+      cli.action.done(`done, v${releases[0].version}`)
+    })
+  )
+}
```
Comment (msakrejda) [inline commands/connection_pooling.js:37]: So this is largely copied from [addons:attach](https://github.com/heroku/heroku-cli-addons/blob/master/commands/addons/attach.js), right? Do you think this is the right approach @jdxcode? The code looks fine to me, but I'm nervous about the copy/pasting (though I don't really have a better suggestion at the moment).

**Review by msakrejda — COMMENTED** [review]
Sorry about the delay. Looks reasonable though I'd love to hear from @jdxcode. Approval pending the checkbox items in the pull request description.

File: commands/connection_pooling.js (hunk 1 above)
Comment (coreypurcell) [reply commands/connection_pooling.js:37]: it is very much just copied from addons:attach.

Comment (jdx) [reply commands/connection_pooling.js:37]: all seems fine to me!

**Review by jdx — APPROVED** [review]

File: commands/connection_pooling.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+const host = require('../lib/host')
+
+function * run (context, heroku) {
+  const util = require('../lib/util')
+  const fetcher = require('../lib/fetcher')(heroku)
+  const {app, args, flags} = context
+
+  let db = yield fetcher.addon(app, args.database)
+  let addon = yield heroku.get(`/addons/${encodeURIComponent(db.name)}`)
+  let credential = `${flags.credential || 'default'}`
```
Comment (msakrejda) [inline commands/connection_pooling.js]: This works but the `flags.credentials` is going to be a string anyway, so you could do `flags.credential || 'default'` without wrapping in a template string.

File: commands/connection_pooling.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  if (util.starterPlan(db)) throw new Error('This operation is not supported by Hobby tier databases.')
+
+  let attachment = yield cli.action(
+    `Enabling Connection Pooling${credential === 'default' ? '' : ' for credential ' + cli.color.addon(credential)} on ${cli.color.addon(addon.name)} to ${cli.color.app(app)}`,
+    heroku.post(`/client/v11/databases/${db.name}/connection-pooling`, {
```
Comment (msakrejda) [inline commands/connection_pooling.js]: Why not `encodeURIComponent(db.name)` here? I mean, I don't think it really matters--I don't believe add-on names support any URL-unsafe characters--but we should be consistent.

File: test/commands/connection_pooling.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      pg.post(`/client/v11/databases/${addon.name}/connection-pooling`, {
+        credential: name
+      }).reply(201, {name: 'HEROKU_COLOR'})
+    })
+
+    it('attaches the pgbouncer url', () => {
+      return cmd.run({app: 'myapp', args: {database: 'postgres-1'}, flags: {credential: name}})
+      .then(() => expect(cli.stdout, 'to equal', ``))
+      .then(() => expect(cli.stderr, 'to contain', 'Setting HEROKU_COLOR config vars and restarting myapp... done, v0\n'))
+    })
```
Comment (msakrejda) [inline test/commands/connection_pooling.js]: Can you add tests for no credential specified so we can validate it defaults to the right thing and for the `--as` flag present, to validate that's passed along correctly?

**Review by msakrejda — CHANGES_REQUESTED** [review]
Some minor comments, but looks great. Sorry it took me so long to follow up. Feel free to ping me in the future.

**Comment by binarycleric** [comment]
Will do. Thanks for the feedback @uhoh-itsmaciek.

File: test/commands/connection_pooling.js (hunk 4 above)
Comment (binarycleric) [reply test/commands/connection_pooling.js]: @uhoh-itsmaciek Done.

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "heroku-pg",
   "description": "Heroku CLI plugin to manage Postgres.",
-  "version": "2.4.2",
+  "version": "2.5.0",
```
Comment (msakrejda) [inline package.json]: The version should be bumped separately and that's done automatically by `np`: see https://github.com/heroku/heroku-pg#publishing

**Review by msakrejda — CHANGES_REQUESTED** [review]
Awesome, thanks! Just drop the `version` change and we're good to go.

**Review by msakrejda — APPROVED** [review]

**Pull request merged**